plugins/module_utils/ec2.py

```diff
@@ -1343,6 +1343,51 @@ def get_sg_id(sg):
     return sec_group_id_list
 
 
+# EC2 Transit Gateway VPC Attachment Error handler
+class EC2TransitGatewayVPCAttachmentErrorHandler(AWSErrorHandler):
+    _CUSTOM_EXCEPTION = AnsibleEC2Error
+
+    @classmethod
+    def _is_missing(cls):
+        return is_boto3_error_code("InvalidGatewayID.NotFound")
+
+
+@EC2TransitGatewayVPCAttachmentErrorHandler.common_error_handler("describe transit gateway attachments")
+@AWSRetry.jittered_backoff()
+def describe_vpc_attachments(
+    client, **params: Dict[str, Union[List[str], bool, List[Dict[str, Union[str, List[str]]]]]]
+) -> List:
+    paginator = client.get_paginator("describe_transit_gateway_vpc_attachments")
+    return paginator.paginate(**params).build_full_result()["TransitGatewayVpcAttachments"]
+
+
+@EC2TransitGatewayVPCAttachmentErrorHandler.common_error_handler("create transit gateway vpc attachment")
+@AWSRetry.jittered_backoff()
+def create_vpc_attachment(
+    client, **params: Dict[str, Union[List[str], bool, List[Dict[str, Union[str, List[str]]]]]]
+) -> Optional[Dict[str, Any]]:
+    result = client.create_transit_gateway_vpc_attachment(**params)
+    return result.get("TransitGatewayVpcAttachment", None)
+
+
+@EC2TransitGatewayVPCAttachmentErrorHandler.common_error_handler("modify transit gateway vpc attachment")
+@AWSRetry.jittered_backoff()
+def modify_vpc_attachment(
+    client, **params: Dict[str, Union[List[str], bool, List[Dict[str, Union[str, List[str]]]]]]
+) -> Optional[Dict[str, Any]]:
+    result = client.modify_transit_gateway_vpc_attachment(**params)
+    return result.get("TransitGatewayVpcAttachment", None)
```

Comment:
```suggestion
    client, transit_gateway_attachment_id: str, **params: Dict[str, Union[List[str], bool, List[Dict[str, Union[str, List[str]]]]]]
) -> Dict[str, Any]:
    return client.modify_transit_gateway_vpc_attachment(**params)["TransitGatewayVpcAttachment"]
```

Reply:
`transit_gateway_attachment_id` is already passed inside params.